### PR TITLE
Update couchbase-server-enterprise to 4.6.0-DP

### PR DESCRIPTION
**Pull request by cgriego**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] The commit message includes the cask’s name and version.

### CODE REVIEW

**Comment by cgriego** [comment]
I wouldn't normally submit a pull request for a developer preview, but 4.6.0-DP is the only version that supports macOS Sierra. 4.5.1 doesn't function on Sierra.

**Comment by reitermarkus** [comment]
Let's leave it at `4.5.1` for versions below Sierra. See https://github.com/caskroom/homebrew-cask/tree/master/doc/cask_language_reference#version-comparisons.

File: Casks/couchbase-server-enterprise.rb [hunk 1]
```diff
@@ -1,8 +1,13 @@
 cask 'couchbase-server-enterprise' do
-  version '4.5.1'
-  sha256 'de014c7c134eb97ff00be6b2e6f5d0da84295ce05bbb7bb3a4d3c747a365cd22'
+  if MacOS.version < :sierra
```
Comment (reitermarkus) [inline Casks/couchbase-server-enterprise.rb]: We usually prefer `<=` for checks, so please use `<= :el_capitan` here.

Comment (cgriego) [inline Casks/couchbase-server-enterprise.rb]: I saw that the majority of checks used `<=`, but I also saw an exception in the r-app cask that matched this use case and the `<` code read like your comment. I've updated to `<=` for consistency.

Comment (reitermarkus) [inline Casks/couchbase-server-enterprise.rb]: Thanks, seems like this was overlooked for `r-app`.

Comment (miccal) [inline Casks/couchbase-server-enterprise.rb]: > Thanks, seems like this was overlooked for `r-app`.

Fixed in https://github.com/caskroom/homebrew-cask/pull/30113.

**Comment by miccal** [comment]
Thank you for your contribution.

**Pull request merged**